src/test/test-nsresource.c

```diff
@@ -0,0 +1,34 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <net/if.h>
+
+#include "nsresource.h"
+#include "tests.h"
+#include "fd-util.h"
+#include "namespace-util.h"
+
+TEST(delegatetap) {
+        int r;
+
+        _cleanup_close_ int userns_fd = userns_acquire_self_root();
+        if (ERRNO_IS_NEG_PRIVILEGE(userns_fd) || ERRNO_IS_NEG_NOT_SUPPORTED(userns_fd))
+                return (void) log_tests_skipped("User namespaces not available");
```

Comment:
Not a matter, but using `log_tests_skipped_errno()` is better to know why it fails.